### PR TITLE
[ADD] business_requirement_crm: New function in 11.0 create BR from Leads

### PR DESCRIPTION
**Pull request by martintorresvm**

### CODE REVIEW

**Comment by martintorresvm** [comment]
Work in progress please don't review

**Comment by martintorresvm** [comment]
cc @Tecnativa

**Comment by martintorresvm** [comment]
@rafaelbn @rafaelbn @pedrobaeza

File: business_requirement_crm/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Tecnativa - Victor M.M. Torres
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Business Requirement CRM',
+    'category': 'Business Requirements Management',
+    'summary': 'Convert Leads to Business Requirement',
+    'version': '11.0.1.1.0',
```
Comment (sergio-teruel) [inline business_requirement_crm/__manifest__.py]: change to 11.0.1.0.0

File: business_requirement_crm/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    'website': 'https://www.elico-corp.com/',
```
Comment (sergio-teruel) [inline business_requirement_crm/__manifest__.py]: Change to https://github.com/OCA/business-requirement

File: business_requirement_crm/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "author": "Elico Corp, Odoo Community Association (OCA)",
```
Comment (sergio-teruel) [inline business_requirement_crm/__manifest__.py]: Change to Tecnativa

File: business_requirement_crm/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[ This file must be max 2-3 paragraphs, and is required. ]
+
+This module extends the functionality of ... to support ...
+and to allow you to ...
+
```
Comment (sergio-teruel) [inline business_requirement_crm/readme/DESCRIPTION.rst]: Fix this file...

File: business_requirement_crm/wizard/crm_lead_convert_to_requirement.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            name = self.env['crm.lead'].browse([lead_id]).name
+            result['lead_id'] = lead_id
+            result['business_requirement'] = name
+        return result
+
+    @api.multi
+    def action_lead_to_business_requirement(self):
+        self.ensure_one()
+        # get the lead to transform
+        lead = self.lead_id
```
Comment (sergio-teruel) [inline business_requirement_crm/wizard/crm_lead_convert_to_requirement.py:65]: You not need this assignament

File: business_requirement_crm/wizard/crm_lead_convert_to_requirement.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        customer_history = self.business_requirement
+        partner_id = lead.handle_partner_assignation()[lead.id]
+        business_requirement = ""
+        # create new business.requirement
+        if customer_history:
+            business_requirement = customer_history
+        elif lead.description:
+            business_requirement = lead.description
+        elif lead.name:
+            business_requirement = lead.name
```
Comment (sergio-teruel) [inline business_requirement_crm/wizard/crm_lead_convert_to_requirement.py]: You can do
vals{'bussines_requirement': customer_history or self.lead_id.description or self.lead_id.name or ''

File: business_requirement_crm/wizard/crm_lead_convert_to_requirement.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        requirement = self.env['business.requirement'].create(vals)
+        # move the mail thread
+        lead.message_change_thread(requirement)
+        # move attachments
+        attachments = self.env['ir.attachment'].search([
+            ('res_model', '=', 'crm.lead'),
+            ('res_id', '=', lead.id)])
+        attachments.write({
+            'res_model': 'project.task',
```
Comment (sergio-teruel) [inline business_requirement_crm/wizard/crm_lead_convert_to_requirement.py]: project.task??

File: business_requirement_crm/wizard/crm_lead_convert_to_requirement.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        return {
+            'name': 'Requirement created',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'view_id': view.id,
+            'res_model': 'business.requirement',
+            'type': 'ir.actions.act_window',
+            'res_id': requirement.id,
+            'context': self.env.context
+        }
```
Comment (sergio-teruel) [inline business_requirement_crm/wizard/crm_lead_convert_to_requirement.py]: You can open a form action more simple:
https://github.com/odoo/odoo/blob/b5a04fd29ff19839e788b2c383e3af74c883c54d/odoo/models.py#L1421

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: business_requirement_crm/__manifest__.py (hunk 3 above)
Comment (nikul-serpentcs) [reply business_requirement_crm/__manifest__.py]: @sergio-teruel Why change author?
FYI  If you added new functionality then change `author`.

Comment (pedrobaeza) [reply business_requirement_crm/__manifest__.py]: Because this has been done by us entirely from scratch

Comment (nikul-serpentcs) [reply business_requirement_crm/__manifest__.py]: @pedrobaeza Thanks For a quick response.

FYI here v8, we have https://github.com/OCA/business-requirement/pull/8 , That's why I am asking you

But Good, Go Ahead with this PR

Comment (pedrobaeza) [reply business_requirement_crm/__manifest__.py]: If you look to both source codes, it has nothing to do with 8.0 one, as it has been totally rethought from the new ground we have made, and that's why you are only seeing 1 commit.

**Comment by martintorresvm** [comment]
Changes done @sergio-teruel

**Comment by martintorresvm** [comment]
@sergio-teruel @pedrobaeza Please review

**Review by pedrobaeza — APPROVED** [review]
I have improved several things:

* Translations
* Prepare method
* Some nomenclature
* Extra tests

**Pull request merged**